sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -0,0 +1,197 @@
+//                    '</svg>',
+//                '</div>',
+            '</li>',
+        '</tpl>',
+        {
+            // TODO: deduplicate this and its copy from the teacher dashboard?
+            getDemonstrationBlocks: function(skill) {
+                var demonstrationsRequired = skill.DemonstrationsRequired,
+                    blocks, blocksLength, blockIndex, lowestBlockIndex;
+
+
+                // start with all demonstrations 1 level below the competency level or greater
+                blocks = Ext.Array.map(skill.demonstrations, function(demonstration) {
+                    return Ext.apply({
+                        counted: demonstration.Level >= 8 // TODO: retrieve the competency level dynamically rather than hard coding to 9
+                    }, demonstration);
+                });
+
+
+                // trim lowest demonstrations
+                while ((blocksLength = blocks.length) > demonstrationsRequired) {
+                    for (blockIndex = 0, lowestBlockIndex = null; blockIndex < blocksLength; blockIndex++) {
+                        if (lowestBlockIndex === null || blocks[blockIndex].Level < blocks[lowestBlockIndex].Level) {
+                            lowestBlockIndex = blockIndex;
+                        }
+                    }
+
+                    Ext.Array.splice(blocks, lowestBlockIndex, 1);
+                }
+                
+                // sort counted demonstrations first
+                Ext.Array.sort(blocks, function(a, b) {
+                    if (a.counted && !b.counted) {
+                        return -1;
+                    }
+
+                    if (!a.counted && b.counted) {
+                        return 1;
+                    }
+
+                    return a.DemonstrationID > b.DemonstrationID ? 1 : -1;
+                });
+
+
+                // add empty blocks
+                while (blocks.length < demonstrationsRequired) {
+                    blocks.push({});
+                }
+                
+                if(blocks[0] == {}) {
+                    debugger;
+                }
+       
+                return blocks;
+            }
+        }
+    ],
+    
+    html: [
+      '<ul class="cbl-competency-panels competencies-unloaded" id="studentDashboardCompetenciesList"></ul>'  
+    ],
+
+    listeners: {
+        scope: 'this',
+        click: {
+            fn: 'onListClick',
+            element: 'el',
+            delegate: '.cbl-competency-panels, .cbl-competency-panels'
+/*
+        },
+        mouseover: {
+            fn: 'onSkillNameMouseOver',
+            element: 'el'
+*/
+        }
+    },
+
+    applyPopover: function(newPopover, oldPopover) {
+        return Ext.factory(newPopover, 'Slate.cbl.widget.Popover', oldPopover);
+    },
+
+    applyContentArea: function(contentArea) {
+        if (!contentArea) {
+            return null;
```

Comment:
null is returned here so that any falsy-value is normalized to null and doesn't generate superfluous calls to updateContentArea. A call to [Ext.factory](http://docs.sencha.com/extjs/5.0/5.0.1-apidocs/#!/api/Ext-method-factory) usually takes the place of the rest of this method but I think there was some additional logic here in the past or I was anticipating that some would be added